core/src/main/scala/org/http4s/rho/bits/PathTree.scala

```diff
@@ -88,13 +88,11 @@ private[rho] object PathTree {
 
       case c @ CodecRouter(_, parser) =>
         Leaf { (req, pathstack) =>
-          RuleExecutor.runRequestRules(req, c.router.rules, pathstack).map{ i =>
-            parser.decode(req, false).run.flatMap(_.fold(e =>
-              e.toHttpResponse(req.httpVersion),
-              { body =>
-                // `asInstanceOf` to turn the untyped HList to type T
-                route.action.act(req, (body :: i).asInstanceOf[T])
-              }))
+          RuleExecutor.runRequestRules(req, c.router.rules, pathstack).flatMap{ i =>
+            parser.parse(req).run.map{ body =>
```

Comment:
I like that idea: being able to pass in a handler while maintaining the
ability to continue the current behavior.
On Jul 1, 2016 7:05 AM, "Jacob Barber" <notifications@github.com> wrote:

> In core/src/main/scala/org/http4s/rho/bits/PathTree.scala
> <https://github.com/http4s/rho/pull/131#discussion_r69304012>:
>
> > @@ -88,13 +88,11 @@ private[rho] object PathTree {
> >
> >        case c @ CodecRouter(_, parser) =>
> >          Leaf { (req, pathstack) =>
> > -          RuleExecutor.runRequestRules(req, c.router.rules, pathstack).map{ i =>
> > -            parser.decode(req, false).run.flatMap(_.fold(e =>
> > -              e.toHttpResponse(req.httpVersion),
> > -              { body =>
> > -                // `asInstanceOf` to turn the untyped HList to type T
> > -                route.action.act(req, (body :: i).asInstanceOf[T])
> > -              }))
> > +          RuleExecutor.runRequestRules(req, c.router.rules, pathstack).flatMap{ i =>
> > +            parser.parse(req).run.map{ body =>
>
> Perhaps instead of my changes, we allow people to optionally pass in an
> error handler during the construction of a CodecRouter. The error handler
> would then be used here in a fold. Basically just like the old way, but
> without a hard-coded failure case.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/http4s/rho/pull/131/files/bd8ce424dca99c7ff786829cc29efc652aaa1b73#r69304012>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACz9msFwt8gYKlnSgfQGAmlHlFBry4I1ks5qRR6LgaJpZM4JC0Ce>
> .
>